src/System.Management.Automation/engine/interpreter/Utilities.cs

```diff
@@ -886,9 +882,6 @@ private StorageInfo CreateStorageInfo()
                     // let others access the storage again
                     Interlocked.Exchange(ref _stores, curStorage);
                 }
-#if !SILVERLIGHT && !CORECLR
-                Thread.EndCriticalRegion();
-#endif
```

Comment:
it seems we should return to ClrFacade.cs `internal static SafeHandle GetSafeProcessHandle(Process process)`